src/main/java/com/adroll/cantor/HLLByteArray.java

```diff
@@ -66,15 +84,33 @@ public HLLByteArray merge(HLLByteArray b) {
         return a;
     }
 
-    public byte[] getShard(int shard_index) {
+    public void write(DataOutput out) {
+        try {
+            byte[] a = null;
+            for (byte[] bd : b_data) {
+                if (bd == null && a == null) {
+                    a = new byte[shard_size];
+                    out.write(a);
+                } else if (bd == null) {
+                    out.write(a);
+                } else {
+                    out.write(bd);
+                }
+            }
+        } catch (Exception e) {
+
```

Comment:
shouldn't we let this bubble up? If `write` throws an exception I would think we would want to know about it. 

If this is catching exceptions from some benign situation that we expect to happen but that does not actually impact the writing, I think we should leave a comment as to what that situation is and catch a more specific exception instance type as opposed to just _all_ `Exception`s.